prism/src/main/java/org/apache/falcon/resource/AbstractExtensionManager.java

```diff
@@ -0,0 +1,165 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.falcon.resource;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.falcon.FalconException;
+import org.apache.falcon.FalconWebException;
+import org.apache.falcon.entity.parser.ProcessEntityParser;
+import org.apache.falcon.entity.parser.ValidationException;
+import org.apache.falcon.entity.v0.Entity;
+import org.apache.falcon.entity.v0.EntityType;
+import org.apache.falcon.entity.v0.feed.Feed;
+import org.apache.falcon.entity.v0.process.Process;
+import org.apache.falcon.extensions.jdbc.ExtensionMetaStore;
+import org.apache.falcon.extensions.store.ExtensionStore;
+import org.apache.falcon.persistence.ExtensionJobsBean;
+import org.apache.hadoop.security.authorize.AuthorizationException;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A base class for managing Extension Operations.
+ */
+public class AbstractExtensionManager extends AbstractSchedulableEntityManager {
+    public static final Logger LOG = LoggerFactory.getLogger(AbstractExtensionManager.class);
+
+    private static final String JOB_NAME = "jobName";
+
+    private static final String EXTENSION_NAME = "extensionName";
+    private static final String FEEDS = "feeds";
+    private static final String PROCESSES = "processes";
+    private static final String CONFIG  = "config";
+    private static final String CREATION_TIME  = "creationTime";
+    private static final String LAST_UPDATE_TIME  = "lastUpdatedTime";
+
+    public static void validateExtensionName(final String extensionName) {
+        if (StringUtils.isBlank(extensionName)) {
+            throw FalconWebException.newAPIException("Extension name is mandatory and shouldn't be blank",
+                    Response.Status.BAD_REQUEST);
+        }
+    }
+
+    public String registerExtensionMetadata(String extensionName, String path, String description) {
+        validateExtensionName(extensionName);
```

Comment:
validation is happening at the prism anyways. so not required to check again over here.